packages/synapse-sdk/src/test/storage.test.ts

```diff
@@ -2463,11 +2461,8 @@ describe('StorageService', () => {
         }
 
         try {
-          await (StorageContext as any).selectRandomProvider(testProviders)
-          assert.fail('Should have thrown error')
-        } catch (error: any) {
-          assert.include(error.message, 'StorageContext selectProviderWithPing failed')
-          assert.include(error.message, 'All 2 providers failed health check')
+          const provider = await (StorageContext as any).selectProviderWithPing(testProviders)
+          assert.isNull(provider)
```

Comment:
I prefer returning null over throwing and catching because catching can hide other errors